api/cluster/controller.go

```diff
@@ -210,6 +215,11 @@ func (c *controller) Deploy(ctx context.Context, modelService *models.Service) (
 		return nil, errors.Wrapf(err, fmt.Sprintf("%v (%s)", ErrUnableToCreateNamespace, modelService.Namespace))
 	}
 
+	err = c.createSecrets(ctx, modelService, projectID)
+	if err != nil {
+		return nil, fmt.Errorf("failed creating secret for deployment %s: %w", modelService.Name, err)
+	}
```

Comment:
During deploy time, we create a secret containing all the MLP secrets beforehand. This secret gets deleted when the model gets undeployed, updated, or when the deployment fails.